src/pages/api/mail/contact.ts

```diff
@@ -1,112 +1,108 @@
-    }
-    try {
-      mailingClient.sendMail(mailData, (err) => {
-        if (err) {
-          console.log(err)
-          res.status(500).json({
-            success: false,
-            body: err,
-          })
-        } else {
-          res.status(200).json({
-            success: true,
-          })
-        }
-      })
-      appendSpreadsheet(sheetsData)
-    } catch (err) {
-      console.log(err)
-      res.status(400).json({
+      const aditionalInformationHeader =
+        discord || telegram || github || projectLink || aditionalInformation
+          ? 'Aditional information'
+          : ''
+      const discordHandle = discord ? `Discord handle: ${discord}` : ''
+      const telegramHandle = telegram ? `Telegram handle: ${telegram}` : ''
+      const githubHandle = github ? `Github Profile: ${github}` : ''
+      const projectLinkText = projectLink ? `Project Link: ${projectLink}` : ''
+      const aditionalInformationText = aditionalInformation
+        ? `Additional Information given by the client: ${aditionalInformation}`
+        : ''
+      const mailData = {
+        from: process.env.GMAIL_ACCOUNT as string,
+        to: process.env.EMAIL_RECIPIENT as string,
+        subject: `Project Request from: ${firstName} | ${projectName}`,
+        text: `Who requested:
+         First Name: ${firstName}
+         E-mail: ${email}
+         Project Name: ${projectName}
+         Project Description: ${projectDescription}
+         Type of services: ${services}
+         Budget Range: ${budget}
+         Deadline(yyyy-mm-dd): ${deadline}
+  
+         ${aditionalInformationHeader}
+         ${discordHandle}
+         ${telegramHandle}
+         ${githubHandle}
+         ${projectLinkText}
+         ${aditionalInformationText}
+         `,
+      }
+      const date = new Date()
+      const sheetsData = {
+        timeStamp: date.toUTCString(),
+        whoRequested: firstName,
+        emailContact: email,
+        projectName: projectName,
+        projectDescription: projectDescription,
+        typeOfServices: services.toString(),
+        budgetRange: budget,
+        deadline: deadline,
+        discordHandle: discord ? discord : '',
+        githubHandle: github ? github : '',
+        telegramHandle: telegram ? telegram : '',
+        projectLink: projectLink ? projectLink : '',
+        aditionalInformationText: aditionalInformation
+          ? aditionalInformation
+          : '',
+      }
+      try {
+        mailingClient.sendMail(mailData, (err) => {
+          if (err) {
+            console.log(err)
+            res.status(500).json({
+              success: false,
+              body: err,
+            })
+          } else {
+            res.status(200).json({
+              success: true,
+            })
+          }
+        })
+        appendSpreadsheet(sheetsData)
+      } catch (err) {
+        console.log(err)
+        res.status(400).json({
+          success: false,
+        })
+      }
```

Comment:
In case of an error in the execution of the endpoint is always good to also send the error message. Getting the message with the correct typing from an error response is a bit tricky, but here's a solution:
```suggestion
      } catch (err) {
        console.log(err)
        let message
        if (err instanceof Error) message = err.message
        else message = String(err)
        res.status(400).json({
          success: false,
          message
        })
      }
```